src/gen_alpha_poc/quaternion.h

```diff
@@ -157,17 +157,18 @@ Vector rotation_vector_from_quaternion(const Quaternion& quaternion) {
     auto q2 = quaternion.GetYComponent();
     auto q3 = quaternion.GetZComponent();
 
-    auto sin_angle_squared = q1 * q1 + q2 * q2 + q3 * q3;
-
-    // Return the rotation vector {0, 0, 0} if provided quaternion is null
-    if (close_to(sin_angle_squared, 0.)) {
-        return Vector{0.0, 0.0, 0.0};
+    Vector n(q1, q2, q3);                           // Vector part of quaternion
+    double n_length = n.Length();                   // magnitude of vector
+    double n_length_squared = n_length * n_length;  // magnitude squared
+
+    // If the length of the vector is less than machine precision,
+    // calculate and return rotation vector using method stable for small angles
+    if (n_length < kEpsilon) {
+        return n * (2. / q0 - 2. / 3. * n_length_squared / (q0 * q0 * q0));
     }
 
-    double sin_angle = std::sqrt(sin_angle_squared);
-    double k = 2. * std::atan2(sin_angle, q0) / sin_angle;
-
-    return {q1 * k, q2 * k, q3 * k};
+    // Otherwise, calculate and return rotation vector using this method
+    return n * 4. * std::atan(n_length / (q0 + std::sqrt(q0 * q0 + n_length_squared))) / n_length;
```

Comment:
We might be able to use a ternary operator here to make things more expressive.

```
    return (
        n_length < kEpsilon
            ? (2. / q0 - 2. / 3. * n_length_squared / (q0 * q0 * q0))
            : 4. * std::atan(n_length / (q0 + std::sqrt(q0 * q0 + n_length_squared))) / n_length
    );
```

Reply:
In my option, this is more difficult to read, but I changed it.

Reply:
I think we should optimize for readability rather than minute code optimization. Honestly, we could have gone either way on this, so please feel free to revert back to original version if you want.